### PR TITLE
[New] Monitor changes to map load status

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements `Monitor changes to map load status` in `Maps` category.
URL to README: https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-MonitorChangesToMapLoadStatus/Shared/Samples/Monitor%20changes%20to%20map%20load%20status

## Linked Issue(s)

- `swift/issues/4175`

## How To Test

* Open the sample to load the map and see the load status displayed on the screen.

## Screenshots

monitor-changes-to-map-toad-status

### CODE REVIEW

File: Shared/Samples/Monitor changes to map load status/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Monitor changes to map load status
+
+Determine the map's load status which can be: `notLoaded`, `failedToLoad`, `loading`, or `loaded`.
```
Comment (zkline101) [inline Shared/Samples/Monitor changes to map load status/README.md:3]: `failedToLoad` is not a case in the `LoadStatus` enum for the Swift API.
```suggestion
Determine the map's load status which can be: `notLoaded`, `failed`, `loading`, or `loaded`.
```

File: Shared/Samples/Monitor changes to map load status/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Knowing the map's load state may be required before subsequent actions can be executed.
+
+## How to use the sample
+
+Open the sample to load the map. The load status will be displayed on screen.
+
+## How it works
+
+The `LoadStatus` is `LOADED` when any of the following criteria are met:
```
Comment (zkline101) [inline Shared/Samples/Monitor changes to map load status/README.md:17]:
```suggestion
The `LoadStatus` is `loaded` when any of the following criteria are met:
```

File: Shared/Samples/Monitor changes to map load status/README.metadata.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+    "category": "Maps",
+    "description": "Determine the map's load status which can be: `notLoaded`, `failedToLoad`, `loading`, or `loaded`.",
```
Comment (zkline101) [inline Shared/Samples/Monitor changes to map load status/README.metadata.json:3]:
```suggestion
    "description": "Determine the map's load status which can be: `notLoaded`, `failed`, `loading`, or `loaded`.",
```

File: Shared/Samples/Monitor changes to map load status/MonitorChangesToMapLoadStatusView.swift [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            .overlay(alignment: .top) {
+                Text("Load Status: \(loadStatusText)")
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
+            }
+            .task {
+                // Listen for load status changes.
+                for await loadStatus in map.$loadStatus {
+                    // Set the load status text when the load status updates.
```
Comment (des12437) [inline Shared/Samples/Monitor changes to map load status/MonitorChangesToMapLoadStatusView.swift:37]:
```suggestion
```
This documentation is not necessary.

Comment (zkline101) [reply Shared/Samples/Monitor changes to map load status/MonitorChangesToMapLoadStatusView.swift:37]: Hmm that is a good question. @yo1995 What is the line for documentation in Samples? I understand adding some doc that might seem unnecessary in certain areas could be helpful for a newcomer but when should we say a piece of documentation won't be helpful for anyone?

Comment (yo1995) [reply Shared/Samples/Monitor changes to map load status/MonitorChangesToMapLoadStatusView.swift:37]: Depending on the audience for a sample. I'd say if a sample is more entry level like this one (or Display map, Show callout, etc.), adding more comments is OK. Samples like utility network and offline workflows, which target advanced users, don't need these trivial comments. They only need to focus on the key logics.

**Review by zkline101 — APPROVED** [review]

**Review by des12437 — APPROVED** [review]

**Pull request merged**